File: backend/src/models/Categoria.js
const { DataTypes } = require('sequelize');
const { sequelize } = require('../config/database');

const Categoria = sequelize.define('Categoria', {
    id: {
        type: DataTypes.INTEGER,
        primaryKey: true,
        autoIncrement: true
    },
    nome: {
        type: DataTypes.STRING,
        allowNull: false
    },
    descricao: {
        type: DataTypes.TEXT
    },
    icone: {
        type: DataTypes.STRING
    },
    cor: {
        type: DataTypes.STRING,
        defaultValue: '#667eea'
    },
    ativo: {
        type: DataTypes.BOOLEAN,
        defaultValue: true
    },
    sla_horas: {
        type: DataTypes.INTEGER,
        defaultValue: 72,
        comment: 'Prazo em horas para resolução'
    }
}, {
    tableName: 'categorias'
});

module.exports = Categoria;

File: python-zeladoria/app/models/categoria.py
from sqlalchemy import Column, Integer, String, Boolean
from sqlalchemy.orm import relationship
from app.database.database import Base

class Categoria(Base):
    __tablename__ = "categorias"

    id = Column(Integer, primary_key=True, index=True)
    nome = Column(String, nullable=False)
    descricao = Column(String)
    icone = Column(String)
    cor = Column(String, default="#667eea")
    sla_horas = Column(Integer, default=72)
    ativo = Column(Boolean, default=True)

    # Relacionamentos
    chamados = relationship("Chamado", back_populates="categoria")
